base/iterator.jl

```diff
@@ -19,6 +19,25 @@ done(e::Enumerate, state) = done(e.itr, state[2])
 
 eltype{I}(::Type{Enumerate{I}}) = Tuple{Int, eltype(I)}
 
+
+immutable EnumerateFrom{I}
+    itr::I
+    start::Int
+end
+
+enumerate(itr, start) = EnumerateFrom(itr, start)
+
+length(e::EnumerateFrom) = length(e.itr) - (e.start - 1)
```

Comment:
Yes, it should. Confused by my own example, I think. Thanks, @mbauman.

I'll close this pull request; fix it; add tests; and, resubmit for others to weigh in on.